### ISSUE TITLE
[v2] WaitUntil tests flapping

### ISSUE THREAD

**Issue by c-kruse** [skupperproject/skupper#1615]
**Describe the bug**
I have noticed several of the _WaitUntil tests are flapping in CI. Ex: https://app.circleci.com/pipelines/github/skupperproject/skupper/3745/workflows/efbaeb90-140a-4e98-b9d3-a7a4d05f2f32/jobs/22837

**How To Reproduce**
run the CI a few times.

**Root Cause**

These tests are non-deterministic. What I suspect is happening is that these test cases set a "timeout" flag to one second. The command WaitUntil method coverts that to int(1) [here](https://github.com/skupperproject/skupper/blob/fd9743cec830d71ed6d05b23eddace93f0f86e83/internal/cmd/skupper/listener/kube/listener_delete.go#L111) to pass to `NewSpinnerWithTimeout`, which sets up a context with a timeout of one second [here](https://github.com/skupperproject/skupper/blob/fd9743cec830d71ed6d05b23eddace93f0f86e83/internal/cmd/skupper/utils/spinner.go#L35). It then calls utils.RetryErrorWithContext with that context and a [hard coded 1 second interval.](https://github.com/skupperproject/skupper/blob/fd9743cec830d71ed6d05b23eddace93f0f86e83/internal/cmd/skupper/utils/spinner.go#L40) `RetryErrorWithContext` sets up a ticker for that 1 second and then sleeps for that interval before either running the test function or watching the context timeout.

Can't say I've got a great suggestion of how to do this nicely without a lot of reorganization. Most likely we should just bump this to 2 seconds and live with it. Perhaps in an ideal world there should be some sort of "retry profile" that gets threaded though everything? i.e. a user profile may be to retry every second for 30 seconds by default, a service profile may be an exponential backoff for longer, and a unit test profile may be without delay repeatedly for 10 seconds.

**Comment by c-kruse** [comment]
@nluaces

**Comment by nluaces** [comment]
@c-kruse thanks for looking into it. I will review what you have seen and give you a response as soon as I can.

**Comment by nluaces** [comment]
Fixed this in #1618, let me know what you think.

**Comment by c-kruse** [comment]
Just noticed another one:
https://app.circleci.com/pipelines/github/skupperproject/skupper/4584/workflows/938b6f63-d6f4-4a83-8456-519a491d25b2/jobs/30003/parallel-runs/0/steps/0-107

```
=== RUN   TestCmdLinkGenerate_WaitUntil
=== RUN   TestCmdLinkGenerate_WaitUntil/secret_is_not_returned
=== RUN   TestCmdLinkGenerate_WaitUntil/the_output_only_contains_the_link
metadata: {}

---
=== RUN   TestCmdLinkGenerate_WaitUntil/bad_format_for_the_output
=== RUN   TestCmdLinkGenerate_WaitUntil/the_output_contains_the_link_and_the_secret
    link_generate_test.go:906: assertion failed: err is not nil
=== RUN   TestCmdLinkGenerate_WaitUntil/the_output_contains_the_link_and_the_secret,_but_it_failed_while_deleting_the_certificate
apiVersion: v1
data:
  ca.crt: Y2E=
  tls.crt: dGxzIGNlcnQ=
  tls.key: dGxzIGtleQ==
kind: Secret
metadata:
  name: link-test
type: kubernetes.io/tls

---
metadata: {}

---
--- FAIL: TestCmdLinkGenerate_WaitUntil (3.30s)
    --- PASS: TestCmdLinkGenerate_WaitUntil/secret_is_not_returned (1.00s)
    --- PASS: TestCmdLinkGenerate_WaitUntil/the_output_only_contains_the_link (0.00s)
    --- PASS: TestCmdLinkGenerate_WaitUntil/bad_format_for_the_output (0.00s)
    --- FAIL: TestCmdLinkGenerate_WaitUntil/the_output_contains_the_link_and_the_secret (1.09s)
    --- PASS: TestCmdLinkGenerate_WaitUntil/the_output_contains_the_link_and_the_secret,_but_it_failed_while_deleting_the_certificate (1.07s)
=== RUN   TestCmdLinkStatus_ValidateInput
```
